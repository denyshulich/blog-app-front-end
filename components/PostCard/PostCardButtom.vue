<template>
    <div class="category-container">
        <p class="categories">
            <nuxt-link v-for="ctg in categories" :key="ctg" class="none-decoration category" to="/">
                {{ ctg.toUpperCase() }}
            </nuxt-link>
        </p>
        <button-blue class="position-button" url="/">
            <svg-icon name="speech-bubble" class="icon-bubble" />
            <div class="comment__number">
                <span> {{ comments }} </span>
            </div>
        </button-blue>
    </div>
</template>

<script>
import ButtonBlue from '../base/ButtonBlue.vue';
export default {
    components: { ButtonBlue },
    props: {
        categories: {
            type: Array,
            required: true
        },
        comments: {
            type: Number,
            required: true
        }
    },
    computed: {}
};
</script>

<style lang="scss" scoped>
.category-container {
    position: relative;
    padding: auto 0;
    padding-left: get-m-vw(38px);
    margin-right: get-m-vw(92px);
    margin-left: get-m-vw(-38px);
    border-top: $border;

    @include mediaSize(tablet) {
        padding-left: get-t-vw(38px);
        margin-right: get-t-vw(92px);
        margin-left: get-t-vw(-38px);
    }
}

.category {
    font-size: get-m-vw(13px);
    color: $colorGrey;

    @include mediaSize(tablet) {
        font-size: get-t-vw(13px);
    }

    &:hover {
        color: black;
    }
}

.category:not(:last-child) {
    &::after {
        padding-right: get-m-vw(5px);
        content: ',  ';

        @include mediaSize(tablet) {
            padding-right: get-t-vw(5px);
        }
    }
}

.position-button {
    position: absolute;
    top: get-m-vw(-20px);
    right: get-m-vw(-80px);
    height: get-m-vw(40px);
    padding: 0 get-m-vw(17px);
    color: $colorGrey;
    border-radius: get-m-vw(100px);
    transition: 0.3s color ease-in-out;

    @include mediaSize(tablet) {
        top: get-t-vw(-20px);
        right: get-t-vw(-80px);
        height: get-t-vw(40px);
        padding: 0 get-t-vw(17px);
        border-radius: get-t-vw(100px);
    }
}

.icon-bubble {
    width: get-m-vw(25px);
    height: get-m-vw(25px);
    color: inherit;

    @include mediaSize(tablet) {
        width: get-t-vw(25px);
        height: get-t-vw(25px);
    }
}

.comment__number {
    display: flex;
    padding-left: get-m-vw(10px);

    @include mediaSize(tablet) {
        padding-left: get-t-vw(10px);
    }

    span {
        margin: auto;
        font-size: get-m-vw(20px);

        @include mediaSize(tablet) {
            font-size: get-t-vw(20px);
        }
    }
}
</style>
