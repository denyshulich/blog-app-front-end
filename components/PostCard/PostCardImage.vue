<template>
    <div class="post__meta-tumb">
        <img :src="image" class="post__one-image" />
        <div class="overlay">
            <nuxt-link to="/" class="overlay__link link-url">
                <div class="link-item__container">
                    <svg-icon name="url" class="post__icon" />
                </div>
            </nuxt-link>
            <nuxt-link v-img="{ src: `${image}` }" to="/" class="overlay__link link-loupe">
                <div class="link-item__container">
                    <svg-icon name="loupe" class="post__icon" />
                </div>
            </nuxt-link>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        image: {
            type: String,
            required: true
        }
    }
};
</script>

<style lang="scss" scoped>
.post__one-image {
    z-index: 1;
    object-fit: cover;
    width: get-m-vw(294px);
    height: max-content;
    max-height: get-m-vw(300px);
    object-position: center;
    transition: 1s;

    @include mediaSize(tablet) {
        width: get-t-vw(340px);
        max-height: get-t-vw(300px);
    }
}

.overlay {
    position: absolute;
}

.post__meta-tumb {
    position: relative;
    overflow: hidden;

    &:hover {
        .overlay {
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
            display: flex;
            background-color: rgba(71, 201, 229, 0.8);
        }

        .post__one-image {
            transform: scale(1.2);
        }
    }
}

.overlay__link {
    width: 50%;
}

.post__icon {
    width: get-m-vw(50px);
    height: get-m-vw(50px);
    color: white;

    @include mediaSize(tablet) {
        width: get-t-vw(50px);
        height: get-t-vw(50px);
    }
}

.link-url,
.link-loupe {
    display: flex;
    align-items: center;
    justify-content: flex-end;

    &:hover {
        .link-item__container {
            position: relative;
            display: flex;
            width: get-m-vw(75px);
            height: get-m-vw(75px);
            background-color: white;
            transition: background-color 0.3s ease-in-out, color 0.3s ease-in-out;
            animation-name: icon-animation;
            animation-duration: 0.6s;

            @include mediaSize(tablet) {
                width: get-t-vw(75px);
                height: get-t-vw(75px);
            }

            .post__icon {
                margin: auto;
                color: black;
            }
        }
    }
}

.link-item__container {
    border-radius: 50%;
    transition: background-color 0.1s ease-in-out, color 0.1s ease-in-out;
}

.link-loupe {
    justify-content: flex-start;

    .link-item__container {
        margin-left: get-m-vw(15px);

        @include mediaSize(tablet) {
            margin-left: get-t-vw(15px);
        }
    }
}

.link-url .link-item__container {
    margin-right: get-m-vw(15px);

    @include mediaSize(tablet) {
        margin-right: get-t-vw(15px);
    }
}

@keyframes icon-animation {
    from {
        top: 5vw;
    }

    to {
        top: 0;
    }
}
</style>
