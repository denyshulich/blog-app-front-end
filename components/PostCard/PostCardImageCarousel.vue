<template>
    <div class="carousel-wrapper">
        <VueSlickCarousel ref="carousel" v-bind="slickOptions">
            <div v-for="src of image" :key="src" class="img-wrapper">
                <img :src="src" />
            </div>
        </VueSlickCarousel>
        <div class="ctm-slick-next ctm-slick-arrow" @click="nextSlide">
            <SvgIcon name="right-arrow" class="ctm-slick-icon" />
        </div>
        <div class="ctm-slick-prev ctm-slick-arrow" @click="prevSlide">
            <SvgIcon name="left-arrow" class="ctm-slick-icon" />
        </div>
    </div>
</template>

<script>
import VueSlickCarousel from 'vue-slick-carousel';
import 'vue-slick-carousel/dist/vue-slick-carousel.css';
import 'vue-slick-carousel/dist/vue-slick-carousel-theme.css';
export default {
    name: 'PostCardImageCarousel',
    components: { VueSlickCarousel },
    props: {
        image: {
            type: Array,
            required: true
        }
    },
    data() {
        return {
            slickOptions: {
                slidesToShow: 1,
                arrows: false,
                autoplay: true,
                autoplaySpeed: 7000,
                dots: true
            }
        };
    },
    methods: {
        prevSlide() {
            this.$refs.carousel.prev();
        },
        nextSlide() {
            this.$refs.carousel.next();
        }
    }
};
</script>

<style lang="scss">
.ctm-slick-arrow {
    position: absolute;
    top: 40%;
}

.ctm-slick-icon {
    width: get-m-vw(20px);
    height: get-m-vw(20px);
    margin: auto;

    @include mediaSize(tablet) {
        width: get-t-vw(20px);
        height: get-t-vw(20px);
    }

    @include mediaSize(desktop) {
        width: get-vw(20px);
        height: get-vw(20px);
    }
}

.ctm-slick-prev,
.ctm-slick-next,
.slick-dots {
    display: none;
}

.carousel-wrapper {
    position: relative;
    overflow: hidden;

    &:hover {
        .slick-dots {
            bottom: 10%;
            display: block;

            li button::before,
            li.slick-active button::before {
                color: transparent;
                opacity: 1;
            }

            li button::before {
                display: inline-block;
                width: get-m-vw(10px);
                height: get-m-vw(10px);
                background-color: transparent;
                border: 4px solid #fff;
                border-radius: 50%;

                @include mediaSize(tablet) {
                    width: get-t-vw(10px);
                    height: get-t-vw(10px);
                }

                @include mediaSize(desktop) {
                    width: get-vw(10px);
                    height: get-vw(10px);
                }
            }

            li.slick-active button::before {
                background-color: #fff;
            }
        }

        .ctm-slick-prev {
            left: 0;
            display: flex;
            width: get-m-vw(55px);
            height: get-m-vw(55px);
            color: white;
            border-radius: 0 50% 50% 0;

            @include mediaSize(tablet) {
                width: get-t-vw(55px);
                height: get-t-vw(55px);
            }

            @include mediaSize(desktop) {
                width: get-vw(55px);
                height: get-vw(55px);
            }

            &:hover {
                background-color: white;

                .ctm-slick-icon {
                    color: black;
                }
            }

            & > .ctm-slick-icon {
                margin-left: get-m-vw(15px);

                @include mediaSize(tablet) {
                    margin-left: get-t-vw(7px);
                }

                @include mediaSize(desktop) {
                    margin-left: get-vw(7px);
                }
            }
        }

        .ctm-slick-next {
            right: 0;
            display: flex;
            width: get-m-vw(55px);
            height: get-m-vw(55px);
            color: white;
            border-radius: 50% 0 0 50%;

            @include mediaSize(tablet) {
                width: get-t-vw(55px);
                height: get-t-vw(55px);
            }

            @include mediaSize(desktop) {
                width: get-vw(55px);
                height: get-vw(55px);
            }

            &:hover {
                background-color: white;

                .ctm-slick-icon {
                    color: black;
                }
            }

            & > .ctm-slick-icon {
                margin-right: get-m-vw(15px);

                @include mediaSize(tablet) {
                    margin-right: get-t-vw(7px);
                }

                @include mediaSize(desktop) {
                    margin-right: get-vw(7px);
                }
            }
        }
    }
}

.img-wrapper img {
    object-fit: cover;
    width: get-m-vw(330px);
    height: max-content;
    max-height: get-m-vw(300px);
    object-position: center;
    margin: auto;
    background-image: linear-gradient(gray 100%, transparent 0);

    @include mediaSize(tablet) {
        width: get-t-vw(300px);
        max-height: get-t-vw(300px);
    }

    @include mediaSize(desktop) {
        width: get-vw(400px);
        max-height: get-vw(300px);
    }
}
</style>
