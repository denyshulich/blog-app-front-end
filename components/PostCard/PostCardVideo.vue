<template>
    <div class="post-card__video-player"><div id="player"></div></div>
</template>

<script>
export default {
    name: 'PostCardVidep',
    props: {
        video: {
            type: [Array, String],
            required: true
        }
    },
    data() {
        return {
            videoId: ''
        };
    },

    mounted() {
        this.getVideoId();
        this.addPlayerVideo();
        window.onYouTubeIframeAPIReady = this.onYouTubeIframeAPIReady;
    },
    methods: {
        getVideoId() {
            this.videoId = this.video.split('=')[1];
        },

        addPlayerVideo() {
            const tag = document.createElement('script');

            tag.src = 'https://www.youtube.com/iframe_api';
            const firstScriptTag = document.getElementsByTagName('script')[0];
            firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
        },

        onYouTubeIframeAPIReady() {
            // eslint-disable-next-line
            const player = new YT.Player('player', {
                height: 'unset',
                width: '100%',
                videoId: this.videoId
            });
        }
    }
};
</script>

<style lang="scss">
.post-card__video-player {
    position: relative;
    width: 100%;
    overflow: hidden;
}

.post-card__video-player::after {
    display: block;
    padding-top: 56.25%;
    content: '';
}

.post-card__video-player iframe {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}
</style>
