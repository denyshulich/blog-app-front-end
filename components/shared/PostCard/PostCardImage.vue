<template>
    <div>
        <div class="post__meta-tumb">
            <img :src="image || image[0]" class="post__one-image" />
            <div class="overlay">
                <div class="overlay__link link-url">
                    <NuxtLink to="/" class="link-item__container">
                        <SvgIcon name="url" class="post__icon" />
                    </NuxtLink>
                    <NuxtLink class="link-item__container_desktop" to="/" />
                </div>
                <div class="overlay__link link-loupe">
                    <div v-img="{ src: `${image || image[0]}` }" class="link-item__container">
                        <SvgIcon name="loupe" class="post__icon" />
                    </div>
                    <div
                        v-img="{ src: `${image || image[0]}` }"
                        class="link-item__container_desktop"
                    />
                </div>
            </div>
        </div>
    </div>
</template>

<script>
import Vue from 'vue';
import VueImg from 'v-img';
const vueImgConfig = {
    altAsTitle: false,
    sourceButton: false,
    openOn: 'click',
    thumbnails: false
};
Vue.use(VueImg, vueImgConfig);

export default {
    name: 'PostCardImage',
    props: {
        image: {
            type: [String, Array],
            required: true
        }
    }
};
</script>

<style lang="scss" scoped>
.post__one-image {
    z-index: 1;
    object-fit: cover;
    width: get-m-vw(280px);
    height: max-content;
    max-height: get-m-vw(300px);
    object-position: center;
    transition: 1s;

    @include mediaSize(tablet) {
        width: get-t-vw(300px);
        max-height: get-t-vw(300px);
    }

    @include mediaSize(desktop) {
        width: get-vw(400px);
        max-height: get-vw(300px);
    }
}

.overlay {
    position: absolute;
}

.post__meta-tumb {
    position: relative;
    overflow: hidden;

    &:hover {
        .overlay {
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
            display: flex;
            background-color: rgba(71, 201, 229, 0.8);
        }

        .post__one-image {
            transform: scale(1.1);
        }
    }
}

.overlay__link {
    width: 50%;
}

.post__icon {
    width: get-m-vw(30px);
    height: get-m-vw(30px);
    color: white;

    @include mediaSize(tablet) {
        width: get-t-vw(30px);
        height: get-t-vw(30px);
    }

    @include mediaSize(desktop) {
        width: get-vw(30px);
        height: get-vw(30px);
    }
}

.link-url,
.link-loupe {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: flex-end;

    .post__icon {
        margin: auto;
    }

    .link-item__container_desktop {
        display: none;

        @include mediaSize(desktop) {
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
            display: block;
            cursor: pointer;
        }
    }

    &:hover {
        .link-item__container {
            &:hover {
                background-color: white;
                box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
                transition: background-color 0.3s ease-in-out, color 0.3s ease-in-out;
                animation-name: icon-animation;
                animation-duration: 0.3s;

                .post__icon {
                    color: black;
                }
            }

            @include mediaSize(desktop) {
                background-color: white;
                box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
                transition: background-color 0.3s ease-in-out, color 0.3s ease-in-out;
                animation-name: icon-animation;
                animation-duration: 0.3s;

                .post__icon {
                    color: black;
                }
            }
        }
    }
}

.link-item__container {
    position: relative;
    display: flex;
    width: get-m-vw(60px);
    height: get-m-vw(60px);
    border-radius: 50%;
    transition: background-color 0.1s ease-in-out, color 0.1s ease-in-out;

    @include mediaSize(tablet) {
        width: get-t-vw(60px);
        height: get-t-vw(60px);
    }

    @include mediaSize(desktop) {
        width: get-vw(60px);
        height: get-vw(60px);
    }
}

.link-loupe {
    justify-content: flex-start;

    .link-item__container {
        // margin-left: get-m-vw(15px);

        @include mediaSize(tablet) {
            margin-left: get-t-vw(15px);
        }

        @include mediaSize(desktop) {
            margin-left: get-vw(15px);
        }
    }
}

.link-url .link-item__container {
    // margin-right: get-m-vw(15px);

    @include mediaSize(tablet) {
        margin-right: get-t-vw(15px);
    }

    @include mediaSize(desktop) {
        margin-right: get-vw(15px);
    }
}

@keyframes icon-animation {
    from {
        top: 1vw;
    }

    to {
        top: 0;
    }
}
</style>
