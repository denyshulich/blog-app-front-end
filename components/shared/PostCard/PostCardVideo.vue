<template>
    <div class="post-card__video-player">
        <LazyYoutubeVideo :src="'https://www.youtube.com/embed/' + getVideoId" />
    </div>
</template>

<script>
import LazyYoutubeVideo from 'vue-lazy-youtube-video';
import 'vue-lazy-youtube-video/dist/style.css';
export default {
    name: 'PostCardVidep',
    components: {
        LazyYoutubeVideo
    },
    props: {
        video: {
            type: [Array, String],
            required: true
        }
    },
    computed: {
        getVideoId() {
            return this.video.split('=')[1];
        }
    }
};
</script>

<style lang="scss">
.post-card__video-player {
    position: relative;
    width: 100%;
    overflow: hidden;
}

.post-card__video-player iframe {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}
</style>
