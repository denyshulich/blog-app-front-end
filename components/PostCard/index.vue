<template>
    <div class="layout">
        <!-- FIXME: Компоненты без контента должны быть самозакрывающимися -->
        <div class="container">
            <div v-if="videoTumbler">
                <PostCardVideo :video="posts.video"></PostCardVideo>
            </div>
            <div v-else>
                <PostCardImageCarousel
                    v-if="imageTumbler"
                    :image="posts.img"
                ></PostCardImageCarousel>
                <post-card-image v-else :image="posts.img"></post-card-image>
            </div>
            <div class="wraper">
                <PostCardSubtitle :avtor="posts.avtor" :date="posts.date"></PostCardSubtitle>
                <PostCardTitle :title="posts.title" :text="posts.text"></PostCardTitle>
                <PostCardButtom
                    :categories="posts.categories"
                    :comments="posts.comments"
                ></PostCardButtom>
            </div>
        </div>
    </div>
</template>

<script>
import PostCardButtom from './PostCardButtom.vue';
import PostCardImage from './PostCardImage.vue';
import PostCardImageCarousel from './PostCardImageCarousel.vue';
import PostCardSubtitle from './PostCardSubtitle';
import PostCardTitle from './PostCardTitle.vue';
import PostCardVideo from './PostCardVideo.vue';

export default {
    components: {
        PostCardImage,
        PostCardSubtitle,
        PostCardImageCarousel,
        PostCardTitle,
        PostCardButtom,
        PostCardVideo
    },
    props: {
        posts: {
            type: Object,
            required: true
        }
    },
    computed: {
        imageTumbler() {
            if (Array.isArray(this.posts.img) && this.posts.img.length > 1) {
                return true;
            } else {
                return false;
            }
        },

        videoTumbler() {
            if (this.posts.video !== undefined) {
                return true;
            } else {
                return false;
            }
        }
    }
};
</script>

<style lang="scss" scoped>
.layout {
    display: flex;
    padding: get-m-vw(15px) 0;

    @include mediaSize(tablet) {
        padding: get-t-vw(15px) 0;
    }

    @include mediaSize(desktop) {
        padding: get-vw(15px) 0;
    }
}

.wraper {
    padding: get-m-vw(32px) get-m-vw(38px) 0 get-m-vw(38px);

    @include mediaSize(tablet) {
        padding: get-t-vw(32px) get-t-vw(38px) 0 get-t-vw(38px);
    }

    @include mediaSize(desktop) {
        padding: get-vw(32px) get-vw(38px) 0 get-vw(38px);
    }
}

.container {
    position: relative;
    top: 0;
    width: get-m-vw(330px);
    height: max-content;
    margin: auto;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25);
    transition: top 0.6s ease-in-out, box-shadow 0.6s ease-in-out;

    @include mediaSize(tablet) {
        width: get-t-vw(300px);
    }

    @include mediaSize(desktop) {
        width: get-vw(400px);
    }

    &:hover {
        top: get-m-vw(1px);
        border: $border;
        box-shadow: none;
        transition: top 0.6s ease-in-out, box-shadow 0.6s ease-in-out;
    }
}
</style>
