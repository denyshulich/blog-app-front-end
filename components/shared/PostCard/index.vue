<template>
    <div class="layout">
        <div class="container">
            <div v-if="videoTumbler">
                <PostCardVideo :video="posts.video" />
            </div>
            <div v-else>
                <PostCardImageCarousel v-if="imageTumbler" :image="posts.img" />
                <PostCardImage v-else :image="posts.img" />
            </div>
            <div class="wraper">
                <PostCardSubtitle :author="posts.author" :date="posts.date" />
                <PostCardTitle :title="posts.title" :text="posts.text" />
                <PostCardBottom :categories="posts.categories" :comments="posts.comments" />
            </div>
        </div>
    </div>
</template>

<script>
import PostCardBottom from './PostCardBottom';
import PostCardImage from './PostCardImage';
import PostCardImageCarousel from './PostCardImageCarousel';
import PostCardSubtitle from './PostCardSubtitle';
import PostCardTitle from './PostCardTitle';
import PostCardVideo from './PostCardVideo';

export default {
    components: {
        PostCardImage,
        PostCardSubtitle,
        PostCardImageCarousel,
        PostCardTitle,
        PostCardBottom,
        PostCardVideo
    },
    props: {
        posts: {
            type: Object,
            required: true
        }
    },
    computed: {
        imageTumbler() {
            if (Array.isArray(this.posts.img) && this.posts.img.length > 1) {
                return true;
            } else {
                return false;
            }
        },

        videoTumbler() {
            if (this.posts.video !== undefined) {
                return true;
            } else {
                return false;
            }
        }
    }
};
</script>

<style lang="scss" scoped>
.layout {
    display: flex;
    padding: get-m-vw(15px) 0;

    @include mediaSize(tablet) {
        padding: get-t-vw(15px) 0;
    }

    @include mediaSize(desktop) {
        padding: get-vw(15px) 0;
    }
}

.wraper {
    padding: get-m-vw(20px) get-m-vw(38px) 0 get-m-vw(38px);

    @include mediaSize(tablet) {
        padding: get-t-vw(20px) get-t-vw(38px) 0 get-t-vw(38px);
    }

    @include mediaSize(desktop) {
        padding: get-vw(20px) get-vw(38px) 0 get-vw(38px);
    }
}

.container {
    position: relative;
    width: get-m-vw(270px);
    margin: 0 auto;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25);
    transition: top 0.6s ease-in-out, box-shadow 0.6s ease-in-out;

    @include mediaSize(tablet) {
        width: get-t-vw(300px);
    }

    @include mediaSize(desktop) {
        width: get-vw(400px);
    }

    &:hover {
        top: get-m-vw(1px);
        border: $border;
        box-shadow: none;
        transition: top 0.6s ease-in-out, box-shadow 0.6s ease-in-out;
    }
}
</style>
