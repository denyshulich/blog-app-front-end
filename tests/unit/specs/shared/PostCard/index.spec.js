import { shallowMount } from '@vue/test-utils';
// import MatchMediaMock from 'jest-matchmedia-mock';
// eslint-disable-next-line
import matchMedia from '~/tests/unit/mocks/match-media';

import PostCard from '~/components/PostCard';
import PostCardImage from '~/components/PostCard/PostCardImage';
import PostCardImageCarousel from '~/components/PostCard/PostCardImageCarousel';
import PostCardVideo from '~/components/PostCard/PostCardVideo';

// let matchMedia;

const postWithImage = {
    img: '',
    avtor: '',
    date: '',
    title: '',
    text: '',
    categories: [],
    comments: 0
};

const postWithSlider = {
    img: ['', ''],
    avtor: '',
    date: '',
    title: '',
    text: '',
    categories: [],
    comments: 0
};

const postWithVideo = {
    img: ['', ''],
    avtor: '',
    date: '',
    title: '',
    text: '',
    categories: [],
    comments: 0,
    video: ''
};

describe('PostCard', () => {
    function createPostCard(posts) {
        const wrapper = shallowMount(PostCard, {
            propsData: {
                posts
            }
            // mocks: {
            //     $device: {
            //         isMobileOrTablet: true
            //     }
            // },
            // stubs: ['NuxtLink', 'SvgIcon']
        });

        return wrapper;
    }

    test('render post card with image', () => {
        const wrapper = createPostCard(postWithImage);

        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.findComponent(PostCardImage).exists()).toBe(true);
    });

    test('render post card with slider', () => {
        const wrapper = createPostCard(postWithSlider);

        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.findComponent(PostCardImageCarousel).exists()).toBe(true);
    });
    test('render post card with slider', () => {
        const wrapper = createPostCard(postWithVideo);

        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.findComponent(PostCardVideo).exists()).toBe(true);
    });
});
