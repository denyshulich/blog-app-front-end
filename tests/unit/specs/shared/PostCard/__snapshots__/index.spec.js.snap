// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard render post card with image 1`] = `
<div
  class="layout"
>
  <div
    class="container"
  >
    <div>
      <post-card-image-stub
        image=""
      />
    </div>
     
    <div
      class="wraper"
    >
      <postcardsubtitle-stub
        avtor=""
        date=""
      />
       
      <postcardtitle-stub
        text=""
        title=""
      />
       
      <postcardbuttom-stub
        categories=""
        comments="0"
      />
    </div>
  </div>
</div>
`;

exports[`PostCard render post card with slider 1`] = `
<div
  class="layout"
>
  <div
    class="container"
  >
    <div>
      <postcardimagecarousel-stub
        image=","
      />
    </div>
     
    <div
      class="wraper"
    >
      <postcardsubtitle-stub
        avtor=""
        date=""
      />
       
      <postcardtitle-stub
        text=""
        title=""
      />
       
      <postcardbuttom-stub
        categories=""
        comments="0"
      />
    </div>
  </div>
</div>
`;

exports[`PostCard render post card with slider 2`] = `
<div
  class="layout"
>
  <div
    class="container"
  >
    <div>
      <postcardvideo-stub
        video=""
      />
    </div>
     
    <div
      class="wraper"
    >
      <postcardsubtitle-stub
        avtor=""
        date=""
      />
       
      <postcardtitle-stub
        text=""
        title=""
      />
       
      <postcardbuttom-stub
        categories=""
        comments="0"
      />
    </div>
  </div>
</div>
`;
